kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -457,8 +462,10 @@ private TransactionCommitResult doCommit(
                           // TODO: handle RemoveFiles.
                           if (!action.isNullAt(ADD_FILE_ORDINAL)) {
                             transactionMetrics.addFilesCounter.increment();
-                            transactionMetrics.addFilesSizeInBytesCounter.increment(
-                                new AddFile(action.getStruct(ADD_FILE_ORDINAL)).getSize());
+                            long addFileSize =
+                                new AddFile(action.getStruct(ADD_FILE_ORDINAL)).getSize();
+                            transactionMetrics.addFilesSizeInBytesCounter.increment(addFileSize);
+                            fileSizeHistogram.ifPresent(histogram -> histogram.insert(addFileSize));
```

Comment:
can we group this a little better? this is a bit easier to read; first we get the size; then we go and increment various things

```java
long addFileSize = ...

transactionMetrics.addFilesCounter.increment();
transactionMetrics.addFilesSizeInBytesCounter.increment(addFileSize);
fileSizeHistogram.ifPresent(histogram -> histogram.insert(addFileSize));
```